p8e-shared/src/main/kotlin/io/provenance/p8e/shared/service/AffiliateService.kt

```diff
@@ -46,6 +50,26 @@ class AffiliateService(
         const val AFFILIATE_INDEX_NAME = "affiliate_index_name"
         const val AFFILIATE_ENCRYPTION_KEY_PAIR = "affiliate_encryption_key_pair"
         const val AFFILIATE_SIGNING_KID = "affiliate_signing_key_id"
+        const val AFFILIATE_ENCRYPTION_PUBLIC_KEY = "affiliate_encryption_public_key"
+    }
+
+    /**
+     *  Return the signer to be used
+     *
+     *  [publicKey] publicKey used to validate
+     *  [signer] The specific signer used to sign contracts
+     */
+    fun getSigner(publicKey: PublicKey): SignerImpl {
+        val affiliateRecord = get(publicKey)
+        return if(affiliateRecord?.privateKey == null) {
+            signerFactory.getSigner(SignerFactoryParam.SmartKeyParam(affiliateRecord?.keyUuid.toString()))
+        } else {
+            signerFactory.getSigner(
+                SignerFactoryParam.PenParam(
+                    KeyPair(affiliateRecord.publicKey.value.toJavaPublicKey(), affiliateRecord.privateKey?.toJavaPrivateKey())
```

Comment:
Can the `privateKey?` become `privateKey`? I think kotlin should have the information to infer that it's non-nullable in this expression based on the `affiliateRecord?.privateKey == null` statement.

Reply:
Looks like it still complains about it, telling me to `!!` it.